### PR TITLE
Rename `src/*/vm/error.py` to `src/*/vm/exceptions.py`

### PR DESCRIPTION
**Pull request by BoboTiG**
### What was wrong?

Related to Issue #479.

### How was it fixed?

Search & replace.

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: src/ethereum/byzantium/vm/exceptions.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 Introduction
 ------------
 
-Errors which cause the EVM to halt exceptionally.
+Exceptions which cause the EVM to halt exceptionally.
```
Comment (BoboTiG) [inline src/ethereum/byzantium/vm/exceptions.py:12]: Maybe the repetition is not good here: "exceptions ... exceptionally".

File: src/ethereum/byzantium/vm/instructions/environment.py [hunk 2]
```diff
@@ -168,7 +168,7 @@ def calldataload(evm: Evm) -> None:
     # Converting start_index to Uint from U256 as start_index + 32 can
     # overflow U256.
     start_index = Uint(pop(evm.stack))
-    value = evm.message.data[start_index : start_index + 32]
+    value = evm.message.data[start_index: start_index + 32]
```
Comment (SamWilsn) [inline src/ethereum/byzantium/vm/instructions/environment.py:171]: Would you mind running `black` and `isort` as described in [`README.md`](https://github.com/ethereum/execution-specs/blob/master/README.md#development):

```bash
$ pip install -e ".[doc,lint,test]"
$ isort src                         # Organizes imports.
$ black src                         # Formats code.
```

**Review by SamWilsn — CHANGES_REQUESTED** [review]
Looks good so far! Just needs to be updated to include muir glacier and istanbul, and needs to run the formatting tools.

**Comment by BoboTiG** [comment]
Updated. And I also adapted docstrings, I missed that from the original commit.

**Pull request merged**